tests/codex/stores/testqueryiterhelper.nim

```diff
@@ -0,0 +1,65 @@
+import std/sugar
+
+import pkg/stew/results
+import pkg/questionable
+import pkg/chronos
+import pkg/datastore/typedds
+import pkg/datastore/sql/sqliteds
+import pkg/codex/stores/queryiterhelper
+import pkg/codex/utils/asynciter
+
+import ../../asynctest
+import ../helpers
+
+proc encode(s: string): seq[byte] =
+  s.toBytes()
+
+proc decode(T: type string, bytes: seq[byte]): ?!T =
+  success(string.fromBytes(bytes))
+
+asyncchecksuite "Test QueryIter helper":
+  var
+    tds: TypedDatastore
+
+  setupAll:
+    tds = TypedDatastore.init(SQLiteDatastore.new(Memory).tryGet())
```

Comment:
It might be useful to use the LevelDB one, just because it is more picky about disposing iterators. It'll help find issues.
There is no in-memory LevelDB datastore. There's a helper called `TempLevelDb` that is used in tests for this. Examples everywhere! :D

Reply:
I prefer such tests to be done in memory only.

Reply:
I haven't looked into this. I don't mind either way. It's just that I noticed using the levelDB for real changes the timing behavior, which has already in the past revealed issues we didn't see. So you can keep the test Sqlite if you want. I'd still suggest you at least run it a few times with the levelDB thing instead, just to see nothing mysterious shows up.